server/api/server.js

```diff
@@ -228,6 +229,67 @@ server.post('/api/trips/newtrip', async (req, res, next) => {
                 },
             },
         });
+
+        res.status(201).json(newtrip);
+    } catch (err) {
+        next(err);
+    }
+});
+
+server.post(`/api/trip/generate-trip`, async (req, res, next) => {
+    const { authorId, name, startDate, endDate, days, locationId, activities } =
+        req.body;
+    try {
+        const validData =
+            authorId !== undefined &&
+            name !== undefined &&
+            days !== undefined &&
+            locationId !== undefined &&
+            activities !== undefined;
+        if (!validData) {
+            next({ status: 422, message: 'Invalid data' });
+        }
```

Comment:
Nit: I would usually say avoid as many negations as possible. So for this it could be
```suggestion
        const invalidData =
            authorId == undefined ||
            name == undefined ||
            days == undefined ||
            locationId == undefined ||
            activities == undefined;
        if (invalidData) {
            next({ status: 422, message: 'Invalid data' });
        }
```

Reply:
makes sense, is there a better way to validate req body information or just check if undefined? all my post req are handled like this.

Reply:
This kind of validation is fine. It doesn't seem like express has a built in way of doing this but people have built libraries that to do this https://express-validator.github.io/docs/. But I wouldn't worry about integrating this